lib/validators/package-validator.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Validates if the component set package size
+ */
+
+import { ComponentSet, GetFileContentType } from '../models';
+import { Validator } from './validator';
+
+const MAX_COMPONENT_SET_FILE_COUNT = 5000;
+const MAX_COMPONENT_SET_SIZE_MB = 100;
+const MAX_CUSTOM_DATA_FILE_COUNT = 1000;
+const MAX_CUSTOM_DATA_SIZE_MB = 20;
+const MB_IN_BYTES = 1 * 1000 * 1000;
+
+interface FileInfo {
+    path: string;
+    size: number;
+}
+
+export class PackageValidator extends Validator {
+    constructor(
+        error: (errorMessage: string) => false,
+        definition: ComponentSet,
+        private filePaths: Set<string>,
+        private getFileContent: GetFileContentType,
+    ) {
+        super(error, definition);
+    }
+
+    async validate(): Promise<void> {
+        const files: FileInfo[] = [];
+        for (let filePath of Array.from(this.filePaths)) {
+            const file = (await this.getFileContent(filePath)) as Buffer | string;
+            files.push({
+                path: filePath,
+                size: file.length,
+            });
+        }
+        this.validateComponentSetFileCount(files);
+        this.validateComponentSetSize(files);
+        this.validateCustomData(files);
+    }
+
+    validateComponentSetFileCount(files: FileInfo[]): void {
+        if (files.length > MAX_COMPONENT_SET_FILE_COUNT) {
+            this.error(
+                `At ${files.length} files, the component set exceeds the total maximum amount of ${MAX_COMPONENT_SET_FILE_COUNT} files.`,
+            );
+        }
+    }
+
+    validateComponentSetSize(files: FileInfo[]): void {
+        const folderSize = this.calculateFolderSize(files);
+        if (folderSize > MAX_COMPONENT_SET_SIZE_MB) {
+            this.error(
+                `At ${folderSize}MB, the component set exceeds the total maximum size of ${MAX_COMPONENT_SET_SIZE_MB}MB.`,
+            );
+        }
+    }
+
+    validateCustomData(files: FileInfo[]): void {
+        const customDataFiles = files.filter((file) => file.path.startsWith('custom/'));
```

Comment:
Actually not sure what we do at the moment in the other places, but should we take case sensitivity into account?

Reply:
I think we really should.

Reply:
Done. Will make sure this works properly in the ecs-backend as well.